src/gtc/numpy/npir_codegen.py

```diff
@@ -14,56 +14,62 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-import functools
 import textwrap
 from dataclasses import dataclass, field
-from typing import Any, Collection, Dict, Tuple, Union
+from typing import Any, Collection, Optional, Set, Tuple, Union
 
+from eve import SymbolTableTrait
 from eve.codegen import FormatTemplate, JinjaTemplate, TemplatedGenerator
-from eve.visitors import NodeVisitor
-from gt4py.definitions import Extent
 from gtc import common
 from gtc.numpy import npir
 
 
 __all__ = ["NpirCodegen"]
 
 
-def op_delta_from_int(value: int) -> Tuple[str, str]:
-    operator = ""
-    delta = str(value)
-    if value == 0:
-        delta = ""
-    elif value < 0:
-        operator = " - "
-        delta = str(abs(value))
-    else:
-        operator = " + "
-    return operator, delta
+def _paren_wrap(s: str) -> str:
+    return f"({s})"
+
+
+def _slice_string(ch: str, offset: int) -> str:
+    return f"{ch}{offset:+d}:{ch.upper()}{offset:+d}" if offset != 0 else f"{ch}:{ch.upper()}"
 
 
 ORIGIN_CORRECTED_VIEW_CLASS = textwrap.dedent(
     """\
-    class ShimmedView:
-        def __init__(self, field, offsets):
-            # use a numpy array here to avoid dimension reducing slicing of storages, which is prohibited and not needed
-            self.field = field.view(np.ndarray)
+    class Field:
```

Comment:
Using the name `Field` here is a bit confusing for me, specially if it has an attribute `self.field`. I would change it (and maybe also rename `self.field` to `self.field_view`